### PR TITLE
Parse q-value of accept header

### PR DESCRIPTION
**Pull request by antiphoton**

### CODE REVIEW

File: cloudflare_worker/src/lib.rs [hunk 1]
```diff
@@ -51,6 +51,16 @@ pub fn serve_preset_content(url: &str) -> JsValue {
     JsValue::from_serde(&response).unwrap()
 }
 
+#[wasm_bindgen(js_name=shouldResponseDebugInfo)]
```
Comment (twifkak) [inline cloudflare_worker/src/lib.rs:54]: nit: s/Response/Respond/

File: sxg_rs/src/media_type.rs [hunk 2]
```diff
@@ -0,0 +1,52 @@
+struct MediaType {
+    weight_milliles: u16,
+    is_sxg_b3: bool,
+}
+
+impl<'a> MediaType {
+    fn new(input: &'a str) -> Self {
+        let input = input.trim();
+        let tokens: Vec<_> = input.split(';').collect();
+        let is_sxg = tokens[0].eq_ignore_ascii_case("application/signed-exchange");
```
Comment (twifkak) [inline sxg_rs/src/media_type.rs:10]: I think you want to trim after splitting on `;` -- `Accept` [allows OWS](https://tools.ietf.org/html/rfc7231#section-5.3.2) there.

File: sxg_rs/src/media_type.rs [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+                    if let Ok(value) = value.parse::<f64>() {
+                        weight_milliles = (value * 1000.0) as u16;
+                    }
+                } else if name == "v" {
+                    version = Some(value);
+                }
+            }
+        }
+        MediaType {
+            is_sxg_b3: is_sxg && version == Some("b3"),
```
Comment (twifkak) [inline sxg_rs/src/media_type.rs:28]: Also add support for the quoted-string form `"b3"`.

File: sxg_rs/src/media_type.rs [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+        let tokens: Vec<_> = input.split(';').collect();
+        let is_sxg = tokens[0].eq_ignore_ascii_case("application/signed-exchange");
+        let mut weight_milliles = 1000;
+        let mut version = None;
+        for param in tokens.iter().skip(1) {
+            let entry: Vec<_> = param.trim().split('=').collect();
+            if entry.len() == 2 {
+                let name = entry[0].trim();
+                let value = entry[1].trim();
+                if name == "q" {
```
Comment (twifkak) [inline sxg_rs/src/media_type.rs:18]: nit: The name "q" is [case-insensitive](https://tools.ietf.org/html/rfc7231#section-5.3.1).

File: sxg_rs/src/media_type.rs [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+                    if let Ok(value) = value.parse::<f64>() {
+                        weight_milliles = (value * 1000.0) as u16;
+                    }
+                } else if name == "v" {
```
Comment (twifkak) [inline sxg_rs/src/media_type.rs:22]: nit: The name "v" is [case-insensitive](https://tools.ietf.org/html/rfc7231#section-3.1.1.1).

Comment (twifkak) [inline sxg_rs/src/media_type.rs:22]: super nit: "v" is actually not a match if it [occurs after "q"](https://tools.ietf.org/html/rfc7231#section-5.3.2). I don't think it's worth the effort to fix that bug, though.

File: cloudflare_worker/src/lib.rs (hunk 1 above)
Comment (antiphoton) [reply cloudflare_worker/src/lib.rs:54]: Fixed in #21.

File: sxg_rs/src/media_type.rs (hunk 2 above)
Comment (antiphoton) [reply sxg_rs/src/media_type.rs:10]: Fixed in #22.

File: sxg_rs/src/media_type.rs (hunk 4 above)
Comment (antiphoton) [reply sxg_rs/src/media_type.rs:18]: Fixin in #22.

File: sxg_rs/src/media_type.rs (hunk 5 above)
Comment (antiphoton) [reply sxg_rs/src/media_type.rs:22]: Fixed in #22.

Comment (antiphoton) [reply sxg_rs/src/media_type.rs:22]: Fixed in #22.

File: sxg_rs/src/media_type.rs (hunk 3 above)
Comment (antiphoton) [reply sxg_rs/src/media_type.rs:28]: Fixed in #22.

**Pull request merged**